### PR TITLE
PW-1657 Transform 3DS2 dialog in a modal

### PR DESCRIPTION
**Pull request by msilvagarcia**

### CODE REVIEW

File: controllers/front/ThreeDSProcess.php [hunk 1]
```diff
@@ -81,6 +81,7 @@ public function postProcess()
                     'message' => "3D secure 2.0 failed, payload details are not found"
                 ]
             ));
+            return;
```
Comment (cyattilakiss) [inline controllers/front/ThreeDSProcess.php]: after the exit is added, this is not required

File: controllers/front/ThreeDSProcess.php [hunk 2]
```diff
@@ -98,6 +99,7 @@ public function postProcess()
                     'message' => '3D secure 2.0 failed'
                 ]
             ));
+            return;
```
Comment (cyattilakiss) [inline controllers/front/ThreeDSProcess.php]: after the exit is added, this is not required

File: controllers/front/ThreeDSProcess.php [hunk 3]
```diff
@@ -112,6 +114,7 @@ public function postProcess()
                     'token' => $result['authentication']['threeds2.challengeToken']
                 ]
             ));
+            return;
```
Comment (cyattilakiss) [inline controllers/front/ThreeDSProcess.php]: after the exit is added, this is not required

File: views/templates/front/payment.tpl [hunk 4]
```diff
@@ -225,7 +227,7 @@
                             type: "POST",
                             url: paymentProcessUrl,
                             data: data,
-                            dataType: "json",
+                            dataType: "text json",
```
Comment (cyattilakiss) [inline views/templates/front/payment.tpl]: Is there any case when we don't return json back but text?

File: views/templates/front/payment.tpl [hunk 5]
```diff
@@ -296,10 +298,7 @@
                             type: "POST",
                             url: threeDSProcessUrl,
                             data: data,
-                            dataType: "json",
-                            done: function(response) {
-                                return response;
-                            }
+                            dataType: "text json"
```
Comment (cyattilakiss) [inline views/templates/front/payment.tpl]: Is there any case when we don't return json back but text?

File: views/templates/front/payment.tpl (hunk 4 above)
Comment (msilvagarcia) [reply views/templates/front/payment.tpl]: The server response type is `text/html`. According to the jQuery docs, adding `text json` tells the library that the response needs to be converted from plain text to JSON. Adding this solved an issue with the 3DS2 modal for me.

File: views/templates/front/payment.tpl (hunk 5 above)
Comment (msilvagarcia) [reply views/templates/front/payment.tpl]: The server response type is `text/html`. According to the jQuery docs, adding `text json` tells the library that the response needs to be converted from plain text to JSON. Adding this solved an issue with the 3DS2 modal for me.

**Pull request merged**